packages/client/src/components/TransactionTable.js

```diff
@@ -58,25 +33,50 @@ const Row = ({ transaction, displayIndex, onView }) => {
 };
 
 const TransactionTable = ({ safeData, transactions = [], className = "" }) => {
-  const modalContext = useModalContext();
+  const { openModal, closeModal } = useModalContext();
 
   const onViewTransaction = (transaction) => {
-    modalContext.openModal(
-      <TransactionDetails
-        safeOwners={safeData.safeOwners}
-        transaction={transaction}
-        onClose={() => modalContext.closeModal()}
-      />
+    const { actionView, signerResponses } = transaction.blockEventData;
+    const signers = {};
+    safeData.safeOwners.forEach((owner) => {
+      let signerStatus = "pending";
+      if (signerResponses?.[owner.address] === 0) {
+        signerStatus = "approved";
+      } else if (signerResponses?.[owner.address] === 1) {
+        signerStatus = "rejected";
+      }
+      signers[owner.address] = signerStatus;
+    });
+    openModal(
+      <TransactionDetailsModal
+        onClose={closeModal}
+        safeData={safeData}
+        // massage data from graffle to work with TransactionDetails
+        transaction={{
+          status: transaction.status,
+          txID: transaction.flowTransactionId,
+          date: transaction.eventDate,
+          signers,
```

Comment:
Looks like the consumers of this signers object use it with Object.keys, consider using an Array instead of an object, it will help simplify the code

Reply:
I was just trying to match the format already used by https://github.com/DapperCollectives/VESSEL/blob/main/packages/client/src/components/Actions/components/Confirmations.js#L8-L14 so was trying to change as little as possible.